tests/base/datamodules.py

```diff
@@ -36,3 +38,47 @@ def val_dataloader(self):
 
     def test_dataloader(self):
         return DataLoader(self.mnist_test, batch_size=32)
+
+
+class MNISTDataModule(LightningDataModule):
+    def __init__(
+        self, data_dir: str = './', batch_size: int = 32, dist_sampler: bool = False
+    ) -> None:
+        super().__init__()
+
+        self.dist_sampler = dist_sampler
+        self.data_dir = data_dir
+        self.batch_size = batch_size
+
+        # self.dims is returned when you call dm.size()
+        # Setting default dims here because we know them.
+        # Could optionally be assigned dynamically in dm.setup()
+        self.dims = (1, 28, 28)
+
+    def prepare_data(self):
+        # download only
+        MNIST(self.data_dir, train=True, download=True, normalize=(0.1307, 0.3081))
```

Comment:
Just to double check, do we need TrialMNIST or MNIST?

Reply:
I did not want to change TrialMNIST structure for this PR as I had to include a DistributedSampler with shuffle